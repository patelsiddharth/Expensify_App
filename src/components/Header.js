import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify App</h1>
        <NavLink to='/' activeClassName='is-active' exact>DashBoard</NavLink>
        <NavLink to='/create' activeClassName='is-active'>Add Expanse</NavLink>
        <NavLink to='/edit' activeClassName='is-active'>Edit Expanse</NavLink>
        <NavLink to='/help' activeClassName='is-active'>Help Page</NavLink>
    </header>
)

export default Header;