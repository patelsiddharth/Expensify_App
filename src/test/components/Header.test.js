import Header from '../../components/Header';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ReactShallowRenderer from 'react-test-renderer/shallow';

test('should render Header correct', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper.find('h1').length).toBe(1);
    
    expect(wrapper.find('h1').text()).toBe('Expensify App');

    expect(toJSON(wrapper)).toMatchSnapshot();
    
    // const renderer = new ReactShallowRenderer();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
})