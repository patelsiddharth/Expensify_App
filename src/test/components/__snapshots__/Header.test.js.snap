// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correct 1`] = `
<header>
  <h1>
    Expensify App
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    DashBoard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Add Expanse
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/edit"
  >
    Edit Expanse
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
    Help Page
  </NavLink>
</header>
`;
